java/src/main/java/org/liberator/ratdriver/control/SafariControl.java

```diff
@@ -1,19 +1,172 @@
 package org.liberator.ratdriver.control;
 
 import org.liberator.ratdriver.preferences.BasePreferences;
+import org.liberator.ratdriver.preferences.SafariPreferences;
+import org.liberator.ratdriver.settings.BaseSettings;
+import org.liberator.ratdriver.settings.SafariSettings;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.safari.SafariDriverService;
+import org.openqa.selenium.safari.SafariOptions;
+
+import java.io.File;
 
 public class SafariControl extends BrowserControl {
 
-    public SafariControl(){}
+    public SafariOptions Options;
+
+    public SafariDriverService.Builder Builder;
+
+    public SafariDriverService Service;
+
+    public SafariControl() {
+    }
 
-    public SafariControl(BasePreferences preferences){}
+    public SafariControl(BasePreferences preferences) {
+    }
 
     public WebDriver StartDriver() {
-        return null;
+        try {
+            System.setProperty("webdriver.safari.driver", BaseSettings.SafariDriverLocation);
+
+            setSafariOptions();
+            setSafariService();
+
+            if (Service != null && Options != null) {
+                Driver = new SafariDriver(Service, Options);
+            } else if (Options == null && Service != null) {
+                Driver = new SafariDriver(Service);
+            } else if (Options != null) {
+                Driver = new SafariDriver(Options);
+            }
+
+            System.out.println("Started safari driver.");
+            return Driver;
+        } catch (Exception ex) {
+            System.out.println("Could not start safari driver.");
+            return null;
+        }
     }
 
     public WebDriver StartDriver(BasePreferences preferences) {
-        return null;
+        try {
+            System.setProperty("webdriver.safari.driver", BaseSettings.SafariDriverLocation);
+            setImportedPreferences((SafariPreferences) preferences);
+
+            setSafariOptions();
+            setSafariService();
+
+            if (Service != null && Options != null) {
+                Driver = new SafariDriver(Service, Options);
+            } else if (Options == null && Service != null) {
+                Driver = new SafariDriver(Service);
+            } else if (Options != null) {
+                Driver = new SafariDriver(Options);
+            }
+
+            System.out.println("Started safari driver.");
+            return Driver;
+        } catch (Exception ex) {
+            System.out.println("Could not start safari driver.");
+            return null;
+        }
+    }
+
+    private void setImportedPreferences(SafariPreferences preferences){
+        if (preferences != null){
+
+        }
+    }
```

Comment:
que??

Reply:
Fixed, but will be more complex once the merge is complete